app/src/org/commcare/activities/connect/ConnectIdDatabaseHelper.java

```diff
@@ -136,4 +140,74 @@ public static void setRegistrationPhase(Context context, ConnectIdTask phase) {
             storeUser(context, user);
         }
     }
+
+    public static void storeAvailableJobs(Context context, List<ConnectJob> jobs) {
+        List<ConnectJob> existingList = getAvailableJobs(context);
+
+        //Delete jobs that are no longer available
+        for (ConnectJob existing : existingList) {
+            boolean stillExists = false;
+            for (ConnectJob newJob : jobs) {
+                if(existing.getID() == newJob.getID()) {
+                    stillExists = true;
+                    break;
+                }
+            }
+
+            if(!stillExists) {
+                getConnectStorage(context, ConnectJob.class).remove(existing.getID());
```

Comment:
how about using `removeAll` instead with a list of ids ? (bulk db operation vs multiple singular ones)

Reply:
[b5d71a0](https://github.com/dimagi/commcare-android/pull/2695/commits/b5d71a0f4dc6f5a999249da1648d4f0991648e92)